script.js

```diff
@@ -1,4 +1,130 @@
+// Project 2: Beat That!
+// Base
+// There are 2 players and players take turns.
+// When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+// The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+// After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+// Pseudocode
+// 1. Prompt the user to click "submit" to roll dice
+// 2. Automatically goes to player 1.
+// 3. There are 2 random dices which rolls different random number from 1 ~ 6.
+// 4. Show the dices number to the player 1 and wait for the player 1 to choose how to arrange the dice to form numbers.
+// 5. Once the player 1 choose, the program will form the new number for player 1.
+// 6. Switch to player 2.
+// 7. Repeat step 3 to 5.
+// 8. Compare which player has the higher final dice number.
+// 9. Declare winner.
+// 10. Program repeat from 1.
+
+// Global variables
+var playerOne;
+var playerOneFirstDice;
+var playerOneSecondDice;
+var playerOneFinalDice;
+var playerTwo;
+var playerTwoFirstDice;
+var playerTwoSecondDice;
+var playerTwoFinalDice;
+var currentGameMode = "playerOneMode1";
+
+// 2 random dices
+var rollDice1 = function () {
+  // 0 ~ 5.99 -> 0 ~ 5 -> 1 ~ 6
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+var rollDice2 = function () {
+  // 0 ~ 5.99 -> 0 ~ 5 -> 1 ~ 6
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+// Main function
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // Default output
+  var outputValue = "";
+
+  // Game mode
+  if (currentGameMode == "playerOneMode1") {
+    // Loop 1
+    // User cannot key in anything except pressing submit button
+    if (input != "") {
+      outputValue = "Please roll the dice by clicking 'Submit'.";
+    } else {
+      // Player 1 mode 1: Roll 2 dices
+      playerOneFirstDice = rollDice1();
+      playerOneSecondDice = rollDice2();
+      console.log(playerOneFirstDice);
+      console.log(playerOneSecondDice);
```

Comment:
it is good to use console log during development but remove them during code submission for clarity and readability of code